src/System.Text.Json/src/System/Text/Json/Writer/JsonWriterHelper.cs

```diff
@@ -0,0 +1,589 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace System.Text.Json
+{
+    internal static partial class JsonWriterHelper
+    {
+        public static readonly byte[] s_newLineUtf8 = Encoding.UTF8.GetBytes(Environment.NewLine);
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool TryWriteIndentation(Span<byte> buffer, int indent, out int bytesWritten)
+        {
+            Debug.Assert(indent % 2 == 0);
```

Comment:
Why is this guaranteed?  The developer doesn't have control over the identation, and if they do, they can't choose an odd value?

Reply:
The implementation of this method assumes (and requires) that `indent` is an even number (since we are writing two spaces at a time here). I added the assert here to ensure if some change causes an odd value to be passed in, that the assert fails since it would require the implementation here to change as well.

> The developer doesn't have control over the identation

Depends on what you mean by control. A caller of `TryWriteIndentation` can accidentally pass in an odd value and I want to prevent that mistake.